addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -764,6 +764,7 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
 
         const labelInputField = document.querySelector(".modal input#o_link_dialog_label_input");
         const linkPreview = document.querySelector(".modal a#link-preview");
+        // FIXME ? How did it work without Serge's patch ?
         assert.strictEqual(labelInputField.value, 'This website',
             "The label input field should match the link's content");
         assert.strictEqual(linkPreview.innerText, 'This website',
```

Comment:
@qsm-odoo The reason why the test works without the fix originated from https://github.com/odoo/odoo/commit/89b0ce0c74458342fea5e1b093e9383f82e2440f They had refactor and improve the handling of zws. So the `zws` in the link are escaped when the selection is made and thus no need to filter these empty characters.

Reply:
> Now that I think about it, instead of checking that the preview is "This website", we should check for "This website" explicitly so this does not happen anymore.
I'm not sure I understood this sentence...

Reply:
> The bug you describe in your commit message is still there in 16.2, so this PR is still needed

I can't reproduce it anymore (Chrome and Firefox tested)... I'll see with you tomorrow

Reply:
I added two assertions that provide more comprehensive test coverage. The previous test only accounted for one way to open the LinkDialog: selecting a part of the link and clicking on the toolbox icon. Another method is to simply click on the link. In this case, a popover appears, allowing link editing via the edit-link icon. In the latter case, the test fails without my fix. See here the value of the input, which indeed contains the 'two zero-width spaces'.
![image](https://github.com/odoo/odoo/assets/22252071/e2b2f679-f8e0-42ad-be40-b3244bf6fe43)

Reply:
> I'm not sure I understood this sentence...

It was supposed to be that: but the `<zws>` is not shown after sending the message
![image](https://github.com/odoo/odoo/assets/10338094/a4db9490-7ac6-4ea3-9004-dca0447156f1)
I would still do it, what do you think? (it can be done only from this PR here, no need to backport in previous versions)

> So the zws in the link are escaped when the selection is made and thus no need to filter these empty characters.
> [...] selecting a part of the link and clicking on the toolbox icon

You mean this is the explicit call to `setSelection(p, 0, p, 1);` in the test that made it work? :grimacing: That sounds so bad... but I guess with your added steps without that `setSelection`, it makes sense :clap:

Reply:
> You mean this is the explicit call to setSelection(p, 0, p, 1); in the test that made it work?

Indeed, upon calling the setSelection() function, an event is triggered, leading to the invocation of the _onSelectionChange() method in OdooEditor.js. Within this method, the _setLinkZws() function is called. In instances where the selection is not collapsed, this function removes the zws https://github.com/odoo/odoo/blob/c5511970dc101c60b728f4142944bf0a1e8bce1e/addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js#L1766-L1773

Reply:
> I would still do it, what do you think? (it can be done only from this PR here, no need to backport in previous versions)

Ok